src/Tribe/Attendee_Repository.php

```diff
@@ -663,4 +688,414 @@ protected function limit_list( $key, $list ) {
+		$this->trigger_create_actions( $attendee, $attendee_data, $ticket );
+
+		// Clear the attendee cache if post_id is provided.
+		if ( ! empty( $this->updates['post_id'] ) ) {
+			$this->attendee_provider->clear_attendees_cache( $this->updates['post_id'] );
+		}
+
+		return $attendee;
+	}
+
+	/**
+	 * Create an attendee object from ticket and attendee data.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $attendee_data  List of attendee data to be saved.
+	 * @param bool  $return_promise Whether to return a promise object or just the ids
+	 *                              of the updated posts; if `true` then a promise will
+	 *                              be returned whether the update is happening in background
+	 *                              or not.
+	 *
+	 * @return array|Tribe__Promise A list of the post IDs that have been (synchronous) or will
+	 *                              be (asynchronous) updated if `$return_promise` is set to `false`;
+	 *                              the Promise object if `$return_promise` is set to `true`.
+	 *
+	 * @throws Tribe__Repository__Usage_Error If the argument types are not set as expected.
+	 */
+	public function update_attendee( $attendee_data, $return_promise = false ) {
+		if ( empty( $attendee_data['attendee_id'] ) ) {
+			throw new Tribe__Repository__Usage_Error( 'You must provide the attendee_id when updating an attendee from the Attendees Repository class' );
+		}
+
+		$this->by( 'id', $attendee_data['attendee_id'] );
+
+		// Set the attendee arguments accordingly.
+		$this->set_attendee_args( $attendee_data );
+
+		// Update the attendee.
+		$saved = $this->save( $return_promise );
+
+		if ( $return_promise ) {
+			$repository = $this;
+
+			return $saved->then( static function() use ( $repository, $attendee_data ) {
+				// Trigger the update actions.
+				$repository->trigger_update_actions( $attendee_data );
+			} );
+		}
+
+		// Trigger the update actions.
+		$this->trigger_update_actions( $attendee_data );
+
+		return $saved;
+	}
+
+	/**
+	 * Set arguments for attendee.
+	 *
+	 * @since TBD
+	 *
+	 * @param array                         $attendee_data List of additional attendee data.
+	 * @param Tribe__Tickets__Ticket_Object $ticket        The ticket object or null if not relying on it.
+	 *
+	 * @throws Tribe__Repository__Usage_Error If the argument types are not set as expected.
+	 */
+	public function set_attendee_args( $attendee_data, $ticket = null ) {
+		$args = [
+			'attendee_id'       => null,
+			'title'             => null,
+			'full_name'         => null,
+			'email'             => null,
+			'ticket_id'         => $ticket ? $ticket->ID : null,
+			'post_id'           => $ticket ? $ticket->get_event_id() : null,
+			'order_id'          => null,
+			'order_attendee_id' => null,
+			'user_id'           => null,
+			'attendee_status'   => null,
+			'price_paid'        => null,
+			'optout'            => null,
+		];
+
+		$args = array_merge( $args, $attendee_data );
```

Comment:
Maybe use `wp_parse_args`?

Reply:
Intention is to only support an array for `$attendee_data`, so the usefulness of `wp_parse_args` is limited greatly here -- I went for a plain `array_merge` as a result of that